[Bug]: No panic recovery in DB.Worker - A panic during a batch import can crash the whole setup

Example stack trace
```
panic: class AFEfbIdHcJbbBGCJIBcAGAIAaEFFAFcCd: shard k0DD71Ycaq26: findNewLocalEntrypoint called on an empty hnsw graph

goroutine 226 [running]:
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).findNewLocalEntrypoint(0xc246ef03c0, {0x1c1f240, 0xc54a958758}, 0x0, 0x0)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/delete.go:538 +0x32d
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*neighborFinderConnector).pickEntrypoint(0xc1fbd0bb38)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/neighbor_connections.go:277 +0xfb
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*neighborFinderConnector).doAtLevel(0xc1fbd0bb38, 0x7ba339621108?)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/neighbor_connections.go:73 +0x65
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*neighborFinderConnector).Do(0xc1fbd0bb38)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/neighbor_connections.go:62 +0x52
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).findAndConnectNeighbors(0x26a0bc0?, 0x600?, 0xc62e44c000?, {0xc62e44c000?, 0x600?, 0x0?}, 0x0?, 0x7b94806a6bc0?, {0x1c1f240, 0xc54a958750})
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/neighbor_connections.go:30 +0x99
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).insert(0xc246ef03c0, 0xc320029900, {0xc62e44c000, 0x600, 0x600})
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/insert.go:190 +0x785
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).Add(0xc246ef03c0, 0x87, {0xc2baae3300, 0x600, 0x63f})
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/insert.go:64 +0x290
github.com/weaviate/weaviate/adapters/repos/db.(*Shard).updateVectorIndexIgnoreDelete(0xc004027860?, {0xc2baae3300?, 0xc8becb4650?, 0xc0040279a8?}, {0xc86f687ee0?, 0x47?, 0xc86f687ee0?})
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/shard_write_put.go:91 +0x51
github.com/weaviate/weaviate/adapters/repos/db.(*objectsBatcher).storeSingleObjectInAdditionalStorage(0xc83e86ad80, {0x1c16398?, 0xc6d54bbcb0?}, 0xc3e4c136c0, {0x0?, 0x0?, 0x0?}, 0x0?)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/shard_write_batch_objects.go:276 +0x95
github.com/weaviate/weaviate/adapters/repos/db.(*DB).worker(0xc0038cc000)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/repo.go:227 +0x8e
created by github.com/weaviate/weaviate/adapters/repos/db.New
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/repo.go:118 +0x5ce
```

Notes to implementer:
- Make sure the fix applies to both HTTP/gRPC
- Make sure the ongoing async importing also reflects those

--- a/adapters/repos/db/index.go
+++ b/adapters/repos/db/index.go
@@ -15,6 +15,7 @@ import (
 	"context"
 	"fmt"
 	"runtime"
+	"runtime/debug"
 	golangSort "sort"
 	"strings"
 	"sync"
@@ -566,6 +567,17 @@ func (i *Index) putObjectBatch(ctx context.Context, objects []*storobj.Object,
 		wg.Add(1)
 		go func(shardName string, group objsAndPos) {
 			defer wg.Done()
+
+			defer func() {
+				err := recover()
+				if err != nil {
+					for pos := range group.pos {
+						out[pos] = fmt.Errorf("an unexpected error occurred: %s", err)
+					}
+					fmt.Printf("panic: %s\n", err)
+					debug.PrintStack()
+				}
+			}()
 			var errs []error
 			if replProps != nil {
 				errs = i.replicator.PutObjects(ctx, shardName, group.objects,

--- a/adapters/repos/db/index.go
+++ b/adapters/repos/db/index.go
@@ -14,6 +14,7 @@ package db
 import (
 	"context"
 	"fmt"
+	"os"
 	"runtime"
 	"runtime/debug"
 	golangSort "sort"
@@ -574,7 +575,7 @@ func (i *Index) putObjectBatch(ctx context.Context, objects []*storobj.Object,
 					for pos := range group.pos {
 						out[pos] = fmt.Errorf("an unexpected error occurred: %s", err)
 					}
-					fmt.Printf("panic: %s\n", err)
+					fmt.Fprintf(os.Stderr, "panic: %s\n", err)
 					debug.PrintStack()
 				}
 			}()

--- a/adapters/repos/db/shard_write_batch_objects.go
+++ b/adapters/repos/db/shard_write_batch_objects.go
@@ -14,6 +14,7 @@ package db
 import (
 	"context"
 	"fmt"
+	"os"
 	"runtime/debug"
 	"sync"
 	"time"
@@ -254,7 +255,7 @@ func (ob *objectsBatcher) storeSingleObjectInAdditionalStorage(ctx context.Conte
 		err := recover()
 		if err != nil {
 			ob.setErrorAtIndex(fmt.Errorf("an unexpected error occurred: %s", err), index)
-			fmt.Printf("panic: %s\n", err)
+			fmt.Fprintf(os.Stderr, "panic: %s\n", err)
 			debug.PrintStack()
 		}
 	}()

--- a/adapters/repos/db/shard_write_batch_objects.go
+++ b/adapters/repos/db/shard_write_batch_objects.go
@@ -13,6 +13,8 @@ package db
 
 import (
 	"context"
+	"fmt"
+	"runtime/debug"
 	"sync"
 	"time"
 
@@ -248,6 +250,15 @@ func (ob *objectsBatcher) shouldSkipInAdditionalStorage(i int) bool {
 func (ob *objectsBatcher) storeSingleObjectInAdditionalStorage(ctx context.Context,
 	object *storobj.Object, status objectInsertStatus, index int,
 ) {
+	defer func() {
+		err := recover()
+		if err != nil {
+			ob.setErrorAtIndex(fmt.Errorf("an unexpected error occurred: %s", err), index)
+			fmt.Printf("panic: %s\n", err)
+			debug.PrintStack()
+		}
+	}()
+
 	if err := ctx.Err(); err != nil {
 		ob.setErrorAtIndex(errors.Wrap(err, "insert to vector index"), index)
 		return

